sandbox-tools/merge-repos/src/syncReposToStaging.ts

```diff
@@ -507,100 +509,222 @@ async function resolveConflictsToTheirs(git: SimpleGit, commitDetails: ICommitDe
     return false;
 }
 
-/**
- * Deletes the local named branch
- * @param git - The SimpleGit instance for the repo to use
- * @param repoName - The configured repo name *(key of the RepoDetails)
- * @param details - The repo details which is used to create the local branch
- * @param forceDelete - Should the branch be force deleted
- */
-async function deleteLocalBranch(git: SimpleGit, repoName: string, details: IRepoDetails, forceDelete?: boolean) {
-    log(`Removing Local branch for ${repoName}...`)
-    let branches = await git.branch().catch(abort(git, "Failed getting branches"));
-    if (branches && branches.branches[details.mergeBranchName]) {
-        // Remove the local branch
-        await git.deleteLocalBranch(details.mergeBranchName, forceDelete).catch(abort(git, `Failed to remove branch for ${repoName} -- ${details.mergeBranchName}`));
-    }
-}
+async function checkFixBadMerges(git: SimpleGit, repoName: string, baseFolder: string, destFolder: string, commitDetails: ICommitDetails, level: number) {
```

Comment:
This is the new recursive logic that compares the `source` repo contents with the `staged` structure, it will
- replace `staged` files that don't match the `source` file size or content with the a copy of the `source` file.
- Copy missing files/folders that don't exist in the `staged` repo but exist in the `source` repo.
- Remove `staged` file/folder(s) that don't exist in the `source`